### PR TITLE
driving license improvements

### PR DESCRIPTION
**Pull request by xuwhite**
This PR makes the driving license configurable and allows it to be co0nfirmed by supporters.

### CODE REVIEW

**Comment by xuwhite** [comment]
@MyIgel pls provide tests ;)

File: src/Controllers/Admin/UserSettingsController.php [hunk 1]
```diff
@@ -87,6 +88,63 @@ public function saveIfsgCertificate(Request $request): Response
...
+        $user = $this->getUser($request);
+        if (!$this->checkDrivingLicense($user)) {
+            throw new HttpNotFound();
+        }
+
+        $this->checkPermission('user.drive.edit', $this->isDriverLicenseSupporter());
+
+        $user = $this->auth->user();
+        $data = $this->validate($request, [
+            'has_car' => 'optional|checked',
```
Comment (MyIgel) [inline src/Controllers/Admin/UserSettingsController.php:109]: That field is not changeable by admins and thus should not  be saved

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: src/Controllers/SettingsController.php [hunk 2]
```diff
@@ -282,7 +286,7 @@ public function saveIfsgCertificate(Request $request): Response
 
     public function saveDrivingLicense(Request $request): Response
     {
-        if (!$this->checkDrivingLicense()) {
+        if (!($this->checkDrivingLicense() || config('driving_license_enabled'))) {
```
Comment (MyIgel) [inline src/Controllers/SettingsController.php:289]:
```suggestion
        if (!config('driving_license_enabled') && !$this->checkDrivingLicense()) {
```

is cleaner in its intend (and checking the config before querying the db is also faster)

File: src/Controllers/SettingsController.php [hunk 3]
```diff
@@ -296,12 +300,14 @@ public function saveDrivingLicense(Request $request): Response
...
         ]);
 
+        if (!$user->license->drive_confirmed) {
+            $user->license->drive_car = (bool) $data['drive_car'];
+            $user->license->drive_3_5t = (bool) $data['drive_3_5t'];
+            $user->license->drive_7_5t = (bool) $data['drive_7_5t'];
+            $user->license->drive_12t = (bool) $data['drive_12t'];
+            $user->license->drive_forklift = (bool) $data['drive_forklift'];
+        }
         $user->license->has_car = (bool) $data['has_car'];
```
Comment (MyIgel) [inline src/Controllers/SettingsController.php:310]: Imho that could be moved above the if but whatever

**Review by MyIgel — CHANGES_REQUESTED** [review]
Just some smaller fixes / tweaks, then its ready to be merged :)

File: src/Controllers/Admin/UserSettingsController.php [hunk 4]
```diff
@@ -32,11 +32,19 @@ public function __construct(
...
     public function certificate(Request $request): Response
     {
-        if (!config('ifsg_enabled')) {
+        if (!config('ifsg_enabled') && !config('driving_license_enabled')) {
             throw new HttpNotFound();
         }
 
-        $this->checkPermission('user.ifsg.edit', $this->isIfsgSupporter());
+        if (
+            !($this->auth->can('user.ifsg.edit')
```
[the remark this answers is not in the document]
Comment (MyIgel) [reply src/Controllers/Admin/UserSettingsController.php]: the fix is not quite what i imagined (adding another indentation would be helpful) but yeah

File: src/Controllers/Admin/UserSettingsController.php [hunk 5]
```diff
@@ -99,7 +157,9 @@ public function settingsMenu(User $user): array
             $menu[url('/users/' . $user->id . '/certificates')] = [
                 'title' => 'settings.certificates',
                 'icon' => 'card-checklist',
-                'permission' => $this->isIfsgSupporter() ? null : 'user.ifsg.edit',
+                'permission' => ($this->isIfsgSupporter() || $this->isDriverLicenseSupporter())
+                    ? null
+                    : ('user.ifsg.edit' || 'user.drive.edit'),
```
Comment (MyIgel) [inline src/Controllers/Admin/UserSettingsController.php:162]: @xuwhite That does not work that way as its resolving to true and thus hides the menu point from shicos.

File: src/Controllers/SettingsController.php [hunk 6]
```diff
@@ -282,7 +286,7 @@ public function saveIfsgCertificate(Request $request): Response
 
     public function saveDrivingLicense(Request $request): Response
     {
-        if (!$this->checkDrivingLicense()) {
+        if (!config('driving_license_enabled') && !$this->checkDrivingLicense()) {
```
Comment (MyIgel) [inline src/Controllers/SettingsController.php:289]: That should be an `||` then

**Review by MyIgel — CHANGES_REQUESTED** [review]

**Review by MyIgel — APPROVED** [review]
Looks good so far i guess :D

**Pull request merged**